File: DSA/DSA07002 NGĂN XẾP 2.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    cin.ignore();
    stack<int> A;

    for (int i = 0; i < n; i++)
    {
        string s;
        getline(cin, s);

        stringstream ss(s);

        string temp;
        ss >> temp;

        if (temp == "PUSH")
        {
            int idk;
            ss >> idk;
            A.push(idk);
        }
        else if (temp == "POP")
        {
            if (!A.empty()) A.pop();
        }
        else
        {
            if (!A.empty())
            {
                cout << A.top();
            }
            else
            {
                cout << "NONE";
            }
            cout << endl;
        }
    }
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    submit();

    return 0;
}

File: DSA_python/DSA07002 NGĂN XẾP 2.py
def submit():
    n = int(input())
    A = list()

    for i in range(n):
        s = input().split()

        if s[0] == "PUSH":
            A.append(s[1])
        elif s[0] == "POP":
            if (len(A) != 0): A.pop()
        else:
            if len(A) != 0:
                print(A[len(A)-1])
            else:
                print("NONE")



submit()
